### PR TITLE
RATESWSX-162: add support for multiple tracking ids

### PR DESCRIPTION
**Pull request by rommelfreddy**
FYI @MartinKuhl

### CODE REVIEW

File: src/Model/Request/SubModel/Head/External/Tracking.php [hunk 1]
```diff
@@ -43,12 +41,50 @@ class Tracking extends AbstractModel
...
+     * @param $provider
+     *
+     * @return Tracking
+     *
+     * @deprecated please use the SubModel "\RatePAY\Model\Request\SubModel\Head\External\Tracking\Id" to set the TrackingId
+     */
+    public function setProvider($provider)
+    {
+        $ids = $this->getIds();
+        if (count($ids[0]) === null) {
```
Comment (eiriarte-mendez) [inline src/Model/Request/SubModel/Head/External/Tracking.php]: count on `null` will throw error on PHP 7.3+

**Review by eiriarte-mendez — CHANGES_REQUESTED** [review]
Hi @rommelfreddy 
there could be some problem using count on newer PHP versions.
it would be nice if you change the runner in `.github/workflows/push.yml`:
`ubuntu-18.04` instead of `ubuntu-latest` to allow other checks to run.

File: src/Model/Request/SubModel/Head/External/Tracking.php [hunk 2]
```diff
@@ -43,12 +41,50 @@ class Tracking extends AbstractModel
      */
     public $admittedFields = [
         'Id' => [
-            'mandatory' => false,
-            'cdata' => true,
+            'mandatory' => true,
+            'multiple' => true,
+            'instanceOf' => 'Head\\External\\Tracking\\Id',
```
Comment (hamail-tahir) [inline src/Model/Request/SubModel/Head/External/Tracking.php:46]: Hi @rommelfreddy
I was testing your changes and found that, Adding 'instanceOf' does not make it backwards compatible. If you test this with the current version of the library the below code works and you can dump and see that the ID is set. but if you switch to your changes you get an error.
"Fatal error:  Uncaught Model exception : Wrong class instance set to 'Id' - 'RatePAY\Model\Request\SubModel\Head\External\Tracking\Id"


```php
$mbArr = new RatePAY\ModelBuilder('Head');
$mbArr->setArray([
    'SystemId' => "Example",
    'Credential' => [
        'ProfileId' => PROFILE_ID,
        'Securitycode' => SECURITYCODE
    ],
    'Meta' => [
        'Systems' => [
            'System' => [
                'Name' => 'Magento 2'
            ]
        ]
    ],
    'External' => [
        'Tracking' => [
            'Id'=> '234',
       ]
    ]
]);
```

File: src/Model/Request/SubModel/Head/External/Tracking.php (hunk 1 above)
Comment (rommelfreddy) [reply src/Model/Request/SubModel/Head/External/Tracking.php]: `$ids[0]`  is also bullsh*t. :) 
fixed the whole line.

File: src/Model/Request/SubModel/Head/External/Tracking.php (hunk 2 above)
Comment (rommelfreddy) [reply src/Model/Request/SubModel/Head/External/Tracking.php:46]: Well, this should be changed by the implementation.
I added the backward compatibility over a little hack.

File: src/Model/Request/SubModel/Head/External/Tracking.php (hunk 1 above)
Comment (rommelfreddy) [reply src/Model/Request/SubModel/Head/External/Tracking.php]: @eiriarte-mendez changed the ubuntu-version in separate PR:
https://github.com/ratepay/php-library/pull/17
(but why? - please see my comment in the PR)

**Review by eiriarte-mendez — APPROVED** [review]

**Review by hamail-tahir — APPROVED** [review]

**Pull request merged**